src/main/resources/data/computercraft/lua/rom/programs/id.lua

```diff
@@ -13,16 +13,32 @@ if sDrive == nil then
     end
 
 else
+    local bAudio = disk.hasAudio(sDrive)
+    if bAudio then
+        print("Audio medium")
+
+        local title = disk.getAudioTitle(sDrive)
+        if title then
+            print("Audio is titled \"" .. title .. "\"")
+        end
+        return
+    end
+
     local bData = disk.hasData(sDrive)
     if not bData then
         print("No disk in drive " .. sDrive)
         return
     end
 
-    print("The disk is #" .. disk.getID(sDrive))
+    local id = disk.getID(sDrive)
+    if id then
+        print("The disk is #" .. id)
+    else
+        print("Non-disk data medium")
```

Comment:
```suggestion
        print("Disk has no id")
```

Reply:
Can be other things then disk. Like computer, or pocket, or turtle. Should we really call it a "Disk" then?

Reply:
Oh, true. Maybe "Drive does not contain a disk"? I'd say wait for #634, but that's not going to give you a localisable string.

Honestly, not entirely opposed to just doing something like the following in the if statement above. `id` isn't meant to be a diagnostic tool for the whole adjacent drive.

```lua
if not disk.hasData(sDrive) or not disk.getID(sDrive) then
  print("No disk in drive " .. sDrive)
  return
end
```

(don't feel like you have to, I'm fine with doing what you're suggesting here).